mlflow/utils/virtualenv.py

```diff
@@ -231,18 +232,57 @@ def _get_virtualenv_name(python_env, work_dir_path, env_id=None):
     )
 
 
-def _create_virtualenv(
-    local_model_path, python_bin_path, env_dir, python_env, extra_env=None, capture_output=False
-):
+def _get_virtualenv_activate_cmd(env_dir: Path) -> str:
     # Created a command to activate the environment
     paths = ("bin", "activate") if not is_windows() else ("Scripts", "activate.bat")
     activate_cmd = env_dir.joinpath(*paths)
-    activate_cmd = f"source {activate_cmd}" if not is_windows() else str(activate_cmd)
+    return f"source {activate_cmd}" if not is_windows() else str(activate_cmd)
+
 
+def _create_virtualenv(
+    local_model_path: Path,
+    python_env: _PythonEnv,
+    env_dir: Path,
+    pyenv_root_dir: Optional[str] = None,
+    env_manager: Literal["virtualenv", "uv"] = em.UV,
+    extra_env: Optional[dict[str, str]] = None,
+    capture_output: bool = False,
+    pip_requirements_override: Optional[list[str]] = None,
+):
+    if env_manager not in {em.VIRTUALENV, em.UV}:
+        raise MlflowException.invalid_parameter_value(
+            f"Invalid value for `env_manager`: {env_manager}. "
+            f"Must be one of `{em.VIRTUALENV}, {em.UV}`"
+        )
+
+    activate_cmd = _get_virtualenv_activate_cmd(env_dir)
     if env_dir.exists():
-        _logger.info("Environment %s already exists", env_dir)
+        _logger.info(f"Environment {env_dir} already exists")
         return activate_cmd
 
+    if env_manager == em.VIRTUALENV:
+        python_bin_path = _install_python(
+            python_env.python, pyenv_root=pyenv_root_dir, capture_output=capture_output
+        )
+        _logger.info(f"Creating a new environment in {env_dir} with {python_bin_path}")
+        env_creation_cmd = [
+            sys.executable,
+            "-m",
+            "virtualenv",
+            "--python",
+            python_bin_path,
+            env_dir,
+        ]
+        install_deps_cmd_prefix = "python -m pip install"
+    elif env_manager == em.UV:
+        _logger.info(
+            f"Creating a new environment in {env_dir} with python "
+            f"version {python_env.python} using uv"
+        )
+        env_creation_cmd = ["uv", "venv", env_dir, f"--python={python_env.python}"]
+        install_deps_cmd_prefix = "uv pip install --prerelease=allow"
+        if _MLFLOW_TESTING:
+            install_deps_cmd_prefix += " -v"
```

Comment:
Do we need this? Can we set this using an environment variable?

Reply:
Yes, let me update to environment variable